collection.go

```diff
@@ -23,38 +31,85 @@ type IndexingPolicy struct {
 }
 
 type IndexingMode string
-
-const (
-	Consistent = IndexingMode("Consistent")
-	Lazy       = IndexingMode("Lazy")
-)
+type OfferThroughput int32
+type OfferType string
+
+//const (
+//	Consistent = IndexingMode("Consistent")
+//	Lazy       = IndexingMode("Lazy")
+//)
+//
+//const (
+//	OfferTypeS1 = OfferType("S1")
+//	OfferTypeS2 = OfferType("S2")
+//	OfferTypeS3 = OfferType("S3")
+//)
 
 type PartitionKey struct {
 	Paths []string `json:"paths"`
 	Kind  string   `json:"kind"`
 }
 
+// https://docs.microsoft.com/en-us/rest/api/cosmos-db/create-a-collection
 type CollectionCreateOptions struct {
 	Id             string          `json:"id"`
 	IndexingPolicy *IndexingPolicy `json:"indexingPolicy,omitempty"`
 	PartitionKey   *PartitionKey   `json:"partitionKey,omitempty"`
+
+	// RTUs [400 - 250000]. Do not use in combination with OfferType
+	OfferThroughput OfferThroughput `json:"offerThroughput,omitempty"`
+	// S1,S2,S3. Do not use in combination with OfferThroughput
+	OfferType OfferType `json:"offerType,omitempty"`
 }
 
 type CollectionReplaceOptions struct {
+	Resource
 	Id             string          `json:"id"`
 	IndexingPolicy *IndexingPolicy `json:"indexingPolicy,omitempty"`
 	PartitionKey   *PartitionKey   `json:"partitionKey,omitempty"`
 }
 
+// https://docs.microsoft.com/en-us/rest/api/cosmos-db/create-a-collection
 func (c *Client) CreateCollection(ctx context.Context, dbName string,
 	colOps CollectionCreateOptions, ops *RequestOptions) (*Collection, error) {
 
-	return nil, ErrorNotImplemented
+	headers := make(map[string]string)
```

Comment:
Did you see the `AsHeaders` methods I used for the other option structs? For all other options I used this method to generate the map of headers.